pvlib/pvsystem.py

```diff
@@ -2537,6 +2553,73 @@ def pvwatts_losses(soiling=2, shading=3, snow=0, mismatch=2, wiring=2,
     return losses
 
 
+def irradiance_loss_pvsyst(effective_irradiance, irradiance_shading_loss,
+                           irradiance_snow_loss, irradiance_soiling_loss):
+    r"""
+    Calculates irradiance losses using a model based on PVsyst.
+
+    The PVsyst method [1]_ linearly reduces the effective irradiance to
+    account for the effects of shading, snow, and soiling. The separate
+    loss factors are compounded to calculate a single loss factor.
+
+    The separate losses are compounded using the following equation:
+
+    .. math::
+
+        L_{total}(\%) = 100 [ 1 - \Pi_i ( 1 - \frac{L_i}{100} ) ]
+
+    :math:L_{total} is the total irradiance loss resulting from the effects of
+    shading, snow, and soiling.
+
+    Note the parameters must each be a series with a DatetimeIndex.  The
+    index of the returned series will be the same as the effective_irradiance
+    index.  All other parameters will be resampled to match this index with
+    a "fill forward" method for filling holes.
+
+    Parameters
+    ----------
+    effective_irradiance : Series
+            The plane-of-array irradiance (W/m2) that is to be reduced.
```

Comment:
Why require a series of effective irradiance when only its DatetimeIndex is required?

Reply:
It's really just another aspect of the broadcasting discussion happening in comments above.  The irradiance_losses percentages should be multiplied by effective_irradiance to reduce the effective_irradiance that goes into the single diode model.  While previous discussion led to the implementation of a percentage output (so the reduction of effective_irradiance happens outside of this function), we still want to make sure the DatetimeIndex of the percentage losses lines up with those of effective_irradiance.  It's true that the only parameter required to do this is the DatetimeIndex of effective_irradiance, but it felt odd to request that explicitly instead of simply pulling it from the Series.  Although requesting it explicitly would allow for better parameter naming which might reduce some of the broadcasting confusion.